lib/insights_cloud/async/connector_playbook_execution_reporter_task.rb

```diff
@@ -138,7 +140,7 @@ def report_job_progress(invocation_status)
         generator = InsightsCloud::Generators::PlaybookProgressGenerator.new(correlation_id)
         all_hosts_success = true
 
-        invocation_status.each do |host_name, status|
+        invocation_status[:hosts_state].each do |host_name, status|
           # skip host if the host already reported that it's finished
           next if status['report_done']
```

Comment:
Not entirely related to the feature, but I think this one can cause problems with serialization too:
```suggestion
          next if ActiveModel::Type::Boolean.new.cast(status['report_done'])
```